base_export_async/static/src/js/data_export.js

```diff
@@ -0,0 +1,89 @@
+odoo.define('base_export_async.DataExport', function(require) {
+    "use strict";
+
+    var core = require('web.core');
+    var DataExport = require('web.DataExport');
+    var framework = require('web.framework');
+    var pyUtils = require('web.py_utils');
+    var Dialog = require('web.Dialog');
+    var _t = core._t;
+
+    DataExport.include({
+        start: function() {
+            this._super.apply(this, arguments);
+            this.async = this.$('#async_export');
+        },
+        export_data: function() {
+            var self = this;
+            if (self.async.is(":checked")) {
+                var exported_fields = this.$(
+                    '.o_fields_list option').map(
+                    function() {
+                        return {
+                            name: (self.records[this.value] ||
+                                this).value,
+                            label: this.textContent ||
+                                this.innerText
+                        };
+                    }).get();
+
+                if (_.isEmpty(exported_fields)) {
+                    Dialog.alert(this, _t(
+                        "Please select fields to export..."
+                    ));
+                    return;
+                }
+                if (!this.isCompatibleMode) {
+                    exported_fields.unshift({
+                        name: 'id',
+                        label: _t('External ID')
+                    });
+                }
+
+                var export_format = this.$export_format_inputs
+                    .filter(':checked').val();
+
+                framework.blockUI();
+                this._rpc({
```

Comment:
can we split/organize the code here as well? :innocent:

Reply:
I copy the original method from Odoo Standard and I just add a small part to send the RPC to another method. Keep it this way may be easier as we understand that this method has been overwritten. If it is really a problem I can rework

Reply:
then mention this at the top of the file and add a comment where you've added your custom code ;)